### PR TITLE
DAS-2044 - Update Swath Projector image and environment variables.

### PR DESCRIPTION
**Pull request by owenlittlejohns**
## Jira Issue ID

DAS-2044

## Description

This PR updates Harmony to use the [new open-source Docker image for the Swath Projector](https://github.com/nasa/harmony-swath-projector/pkgs/container/harmony-swath-projector). I also have taken the chance to update the name to "Swath Projector" everywhere, as the "swot-repr" name was deprecated a while back. Those environment variable changes are a little more pesky, but I'm also about to push a PR to the CI/CD repo updating the Terraform code there, too.

For what it's worth, I think the other projects being migrated mostly have variables names that are still accurate (like HyBIG and MaskFill), so those migrations should just be a case of swapping out the Docker image location.

## Local Test Steps

* Pull this branch.
* Pull the [new open-source Docker image](https://github.com/nasa/harmony-swath-projector/pkgs/container/harmony-swath-projector) locally: `docker pull ghcr.io/nasa/harmony-swath-projector:latest`
* Update your `LOCALLY_DEPLOYED_SERVICES` to replace "swot-repr" (or "swath-reproject") with "swath-projector".
* Rebuild the versions of the local Harmony images you are using: `nvm use && npm i && npm run build-all`
* Start a local Harmony instance: `./bin/bootstrap-harmony`.
* Start up a Jupyter notebook server and run through the [Swath Projector regression test suite](https://github.com/nasa/harmony-regression-tests/tree/main/test/swath-projector) after setting `harmony_host_url='http://localhost:3000` in the notebook. The tests should run and they should still pass.

## PR Acceptance Checklist
* [x] Acceptance criteria met
* [x] Tests added/updated (if needed) and passing
* ~~Documentation updated (if needed)~~

### CODE REVIEW

File: services/harmony/env-defaults [hunk 1]
```diff
@@ -398,13 +398,13 @@ HARMONY_REGRIDDER_LIMITS_MEMORY=512Mi
...
-SWOT_REPROJECT_LIMITS_CPU=128m
-SWOT_REPROJECT_LIMITS_MEMORY=512Mi
-SWOT_REPROJECT_INVOCATION_ARGS='python swotrepr.py'
-SWOT_REPROJECT_SERVICE_QUEUE_URLS='["sds/swot-reproject:latest,http://localstack:4566/queue/swot-reproject.fifo"]'
+SWATH_PROJECTOR_IMAGE=ghcr.io/nasa/harmony-swath-projector:latest
+SWATH_PROJECTOR_REQUESTS_CPU=128m
+SWATH_PROJECTOR_REQUESTS_MEMORY=128Mi
+SWATH_PROJECTOR_LIMITS_CPU=128m
+SWATH_PROJECTOR_LIMITS_MEMORY=512Mi
+SWATH_PROJECTOR_INVOCATION_ARGS='python -m swath_projector'
```
Comment (owenlittlejohns) [inline services/harmony/env-defaults:406]: The Swath Projector repository (and Docker image) was restructures in [this commit](https://github.com/nasa/harmony-swath-projector/commit/da7b2b56c43408fee3a7b48fb2de7d6f3a1c73ff). This now aligns with the [guidelines](https://wiki.earthdata.nasa.gov/display/HARMONY/Harmony+Services+General+Repository+Structure) for a backend service repository. The important thing is it's all now in a `swath_projector` subdirectory, which has a `__main__.py` that is now executed when the service is called.

**Review by flamingbear — APPROVED** [review]
This is great.  I was able to perform all of the tests successfully.  

I did find some things that might have missed being renamed.  You may want to rename some or all of them:

in services/harmony/public/notebook-example.html 

``` shell
13626:  - name: sds/swot-reproject
13632:        template: swot-reproject
13633:        image: !Env ${SWOT_REPR_IMAGE}
13634:        image_pull_policy: !Env ${SWOT_REPR_IMAGE_PULL_POLICY}
13637:          STAGING_PATH: public/sds/swot-reproject
17806: <h2 id="Access-Level-2-swath-regridded-data">Access Level 2 swath regridded data<a class="anchor-link" href="#Access-Level-2-swath-regridded-data">&#182;</a></h2><p>Moving outside of the <code>harmony/gdal</code> service, we will now request regridding from the <code>sds/swot-reproject</code> servic 
```

In services/harmony/test/workflow-orchestration.ts:

``` shell
251:        it('finds a swot-reprojection service work item and can complete it', async function () {
369:    describe('when the first swot-reprojection service work item fails with an error message', function () {
406:      it('does not find any further swot-reproject work', async function () {              
458:    describe('when the first swot-reprojection service work item fails and does not provide an error message', function () {
605:          // sanity check that 3 swot-reproject items were generated by the first query-cmr task
622:          // sanity check that 2 more swot-reproject items were generated by the second query-cmr task
```

**Review by chris-durbin — APPROVED** [review]
Successfully verified in my own sandbox environment.

**Comment by owenlittlejohns** [comment]
Just a quick note here:

The `LOCALLY_DEPLOYED_SERVICES` environment variables will need to be updated in the UAT and production Bamboo deployment environment when those get deployed with this change. (I updated SIT as it deploys automatically after merges)

**Pull request merged**